comtypes/client/_code_cache.py

```diff
@@ -48,23 +48,23 @@ def _find_gen_dir():
         # check type of executable image to determine a subdirectory
         # where generated modules are placed.
         ftype = getattr(sys, "frozen", None)
-        version_str = "%d%d" % sys.version_info[:2]
+        version_str = f"{sys.version_info[0]:d}{sys.version_info[1]:d}"
         if ftype == None:
             # Python script
-            subdir = r"Python\Python%s\comtypes_cache" % version_str
+            subdir = f"Python\\Python{version_str}\\comtypes_cache"
```

Comment:
It might be clearer to decompose the [`sys.version_info`](https://docs.python.org/3.13/library/sys.html#sys.version_info) into variables reflecting it as a tuple of *major*, *minor*, *micro*, *releaselevel*, and *serial*.
```diff
-        version_str = f"{sys.version_info[0]:d}{sys.version_info[1]:d}"
+        pymaj, pymin = sys.version_info[:2]
         if ftype == None:
             # Python script
-            subdir = f"Python\\Python{version_str}\\comtypes_cache"
+            subdir = f"Python\\Python{pymaj:d}{pymin:d}\\comtypes_cache"
```